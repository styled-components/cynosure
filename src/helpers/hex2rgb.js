// @flow

import hexToRGB from '../internalHelpers/_hexToRGB'

/**
 * Convert HEX To RGB.
 *
 * @example
 * // Styles as object usage
 *
 * const styles = {
 *   ...color: hex2rgb('#f00')
 * }
 *
 *
 * // styled-components usage
 * const div = styled.div`
 *   color: ${hex2rgb('#f00')};
 *`
 *
 * // CSS in JS Output
 *
 * div {
 *   color: rgb((255, 0, 0));
 * }
 */

function hex2rgb(hex: string) :any {
  const color = hexToRGB(hex)
  color.splice(-1, 1)
  const str = color.join(', ')
  return `rgb(${str})`
}
export default hex2rgb
