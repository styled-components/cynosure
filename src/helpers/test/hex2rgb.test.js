// @flow
import hex2rgb from '../hex2rgb'

describe('hex2rgb', function() {
  it('should convert hex color to rgb', function() {
    expect({ color: hex2rgb('#f00') }).toMatchSnapshot()
  })

  it('should throw an error if color is invalid', () => {
    expect(() => ({ color: hex2rgb('#test') }))
      .toThrow('Please enter a valid colors')
  })

  it('should throw an error if color is missing', () => {
    expect(() => ({ color: hex2rgb() }))
      .toThrow('Please enter a valid colors')
  })
})
