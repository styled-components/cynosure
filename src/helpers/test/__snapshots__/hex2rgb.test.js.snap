exports[`hex2rgb should convert hex color to rgb 1`] = `
Object {
  "color": "rgb(255, 0, 0)",
}
`;
